lib/web/websocket/connection.js

```diff
@@ -176,9 +177,18 @@ function establishWebSocketConnection (url, protocols, ws, onEstablish, options)
       //    the WebSocket Connection_.
       const secProtocol = response.headersList.get('Sec-WebSocket-Protocol')
 
-      if (secProtocol !== null && secProtocol !== request.headersList.get('Sec-WebSocket-Protocol')) {
-        failWebsocketConnection(ws, 'Protocol was not set in the opening handshake.')
-        return
+      if (secProtocol !== null) {
```

Comment:
if we sent at least one subprotocol, but the server responded with a null header or did not send a header, then according to the specification we should fail too

Reply:
the spec also says

```
6.  If the response includes a |Sec-WebSocket-Protocol| header field
       and this header field indicates the use of a subprotocol that was
       not present in the client's handshake (the server has indicated a
       subprotocol not requested by the client), the client MUST _Fail
       the WebSocket Connection_.
```

the "if the response includes" part seems to contradict what it later says?

Reply:
I see, protocols are optional so the server doesn't need to send (if none are sent from the client). Definitely could be worded better...

Reply:
Yes, protocols are optional. But, if the client specifies them, then this becomes a mandatory requirement for establishing a connection. And if the server does not support any of the client protocols, then it responds with a null header or does not specify one to let the client know that it is not ready for further interaction

Reply:
I completely misunderstood what the rfc was stating.